lib/Echidna/UI.hs

```diff
@@ -78,25 +80,46 @@ ui :: ( MonadCatch m, MonadRandom m, MonadReader x m, MonadUnliftIO m
 ui v w ts d txs = do
   let d' = fromMaybe defaultDict d
   let getSeed = view $ hasLens . to seed . non (d' ^. defSeed)
-  bc <- liftIO $ newBChan 100
   ref <- liftIO $ newIORef defaultCampaign
-  dash <- liftM2 (&&) isTerminal $ view (hasLens . dashboard)
-  timeoutSeconds <- (* 1000000) . fromMaybe (-1) <$> view (hasLens . maxTime)
   let updateRef = use hasLens >>= liftIO . atomicWriteIORef ref
-  let updateUI e = when dash $ readIORef ref >>= writeBChan bc . e
-  waitForMe <- liftIO newEmptyMVar
-  ticker <- liftIO $ forkIO -- update UI every 100ms, instant exit when no UI
-    (when dash $ forever $ threadDelay 100000 >> updateUI CampaignUpdated)
-  _ <- forkFinally -- run worker
-    (void $ timeout timeoutSeconds (campaign updateRef v w ts d txs) >>= \case
-       Nothing -> liftIO $ updateUI CampaignTimedout
-       Just _ -> liftIO $ updateUI CampaignUpdated
-    )
-    (const $ liftIO $ killThread ticker >> putMVar waitForMe ())
-  app <- customMain (mkVty defaultConfig) (Just bc) <$> monitor
-  liftIO $ if dash
-             then void $ app (defaultCampaign, Uninitialized)
-             else takeMVar waitForMe
-  final <- liftIO $ readIORef ref
-  liftIO . putStrLn =<< view (hasLens . finished) <*> pure final <*> getSeed
-  return final
+  timeoutSeconds <- (* 1000000) . fromMaybe (-1) <$> view (hasLens . maxTime)
```

Comment:
Why we had this constant? can you name it?

Reply:
`timeout` function takes time as microseconds, maxTime is seconds. Will make that clear